Replace deprecated onKeyPress with onKeyDown

jenweber opened the pull request:
> What was wrong/why is this fix needed? (quick summary only)

onKeyPress is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event), but it was being used in a checkbox example on the [checkbox role page](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role).

The article guides devs to handle spacebar inputs, however the given example fired for all keys that keyDown would have captured (any alphabet key, enter, spacebar, etc).

Lastly the example was missing focus styles, which are [critical for accessibility](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-visible.html). 

> Anything else that could help us review it

The onKeyPress page suggests using [onKeyDown](https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event) instead, so that's what I used here.

I relied on `event.keyCode` instead of `event.key` because the value of event.key for spacebar is an empty string. Whitespace can be ambiguous and a source of invisible characters pain for JS developers, so I used the unambiguous keycode value instead.

Here's a [codepen](https://codepen.io/jweber/pen/ZEyGPab) showing the new version. To test it, click the checkbox a few times, try tabbing to it via keyboard, and press spacebar a few times.

## Possible followups

I'd love to use a series of if...if else...else statements instead of if + a case switch, to improve readability. If that seems ok to you, I can make that change or open a second PR. I didn't want to change too much at once.

Here's an example refactor:
```js
function changeCheckbox(keyCode) {
  let spacebarKeyCode = 32;
  let item = document.getElementById('chkPref');
  let checked = item.getAttribute('aria-checked');

  if (keyCode && keyCode !== spacebarKeyCode) {
    return;
  } else if (checked === 'true') {
    item.setAttribute('aria-checked', 'false')
  } else {
    item.setAttribute('aria-checked', 'true')
  }
}
```

jenweber committed:
Replace deprecated onKeyPress with onKeyDown:

Also add more a11y guidance to checkbox role docs.

sideshowbarker commented:
> Oops, I got my commit message details backwards, lol. I will fix that later today.

@jenweber It would be great if you could go ahead and make that change

Elchi3 commented:
@jenweber are you planning to come back to this PR?

jenweber commented:
Hi! Thanks for your patience. I have rebased and applied my changes to the new markdown file. I also fixed my git commit message content.

Elchi3 commented:
Thank you, @jenweber! 
@estelle want to review this?

wbamberg reviewed (commented):
Sorry to be so slow to review this. It looks like a great improvement. I had a couple of really minor comments.

But I'd be totally happy to see your refactor with `if/else` in this PR!

jenweber commented:
Thanks for the review! I will aim to get the iterations up within the next week.

jenweber committed:
Fix indentation, use if instead of case switch

jenweber committed:
Fix semicolons

jenweber commented on files/en-us/web/accessibility/aria/roles/checkbox_role/index.md:
```diff
@@ -95,7 +99,11 @@ The following example creates an otherwise non-semantic checkbox element using C
+function changeCheckbox(keyCode) {
+    const spacebarKeyCode = 32;
+    if (keyCode && keyCode !== spacebarKeyCode) {
+      return;
```
Indentation updated. I ❤️ Prettier!

jenweber commented on files/en-us/web/accessibility/aria/roles/checkbox_role/index.md:
```diff
@@ -95,7 +99,11 @@ The following example creates an otherwise non-semantic checkbox element using C
+    if (keyCode && keyCode !== spacebarKeyCode) {
+      return;
+    }
     let item = document.getElementById('chkPref');
```
All updated to use const.

wbamberg reviewed (approved):
Thank you @jenweber , this looks great!

<strike>CI is failing but that's not your fault so I think we can merge this.</strike> apparently it isn't any more

jenweber commented:
Thank you! I rebased on `main` to get CI going again. For what it's worth, I just updated the codepen as well in case you wanted to click test the new JS using ifs and consts.
